src/write/js/js-output-community-repo.ts

```diff
@@ -0,0 +1,147 @@
+import { GeneratorOutput } from "../../types";
+import path from "node:path";
+import fs from "node:fs/promises";
+import { exec as execCb } from "node:child_process";
+import { promisify } from "node:util";
+import { Formatter } from "../formatter";
+
+const exec = promisify(execCb);
+
+const COMMUNITY_NODE_STARTER_REPO_URL = "https://github.com/n8n-io/n8n-nodes-starter.git";
+const MAIN_INPUT_TYPE = "NodeConnectionType.Main";
+
+export async function generateCommunityRepository(data: GeneratorOutput, outputDir: string, repoName: string) {
+  await cloneCommunityStarterRepo(outputDir, repoName);
+  await cleanDirs(data, outputDir, repoName);
+
+  const className = getNodeClassName(data);
+
+  const { Formatter } = await import("~/write/formatter");
+  const formatter = new Formatter();
+
+  const nodeDistFilePath = await writeNodeFile(data, formatter, outputDir, repoName, className);
+  const credDistFilePaths = await writeCredentialFiles(data, formatter, outputDir, repoName);
+
+  await updatePackageJson(formatter, outputDir, repoName, nodeDistFilePath, credDistFilePaths);
+}
+
+async function cloneCommunityStarterRepo(outputDir: string, repoName: string) {
+  await fs.mkdir(`${outputDir}/${repoName}`, { recursive: true });
+
+  await exec(`git clone ${COMMUNITY_NODE_STARTER_REPO_URL} .`, {
+    cwd: `${outputDir}/${repoName}`,
+  });
+}
+
+async function cleanDirs(data: GeneratorOutput, outputDir: string, repoName: string) {
+  await exec(`rm -rf ./${repoName}/nodes/*`, { cwd: outputDir });
+
+  if (!data.credTypes.length) {
+    await exec(`rm -rf ./${repoName}/credentials`, { cwd: outputDir });
+  } else {
+    await exec(`rm -rf ./${repoName}/credentials/*`, { cwd: outputDir });
+  }
+}
+
+function getNodeClassName(data: GeneratorOutput) {
+  return data.nodeType.fileName.split(".")[0];
+}
+
+async function writeNodeFile(
+  data: GeneratorOutput,
+  formatter: Formatter,
+  outputDir: string,
+  repoName: string,
+  className: string,
+) {
+  const nodeDir = path.join(`${outputDir}/${repoName}/nodes/${className}`);
+  await fs.mkdir(nodeDir, { recursive: true });
+
+  const nodeTypePath = path.join(
+    `${outputDir}/${repoName}/nodes/${className}`,
+    data.nodeType.fileName.replace(".json", ".ts"),
+  );
+
+  let source = data.nodeType.source;
+  source = source.replace('"inputs": ["main"],', `inputs: [${MAIN_INPUT_TYPE}],`);
+  source = source.replace('"outputs": ["main"],', `outputs: [${MAIN_INPUT_TYPE}],`);
+
+  let fileContent = `
+    import { INodeType, INodeTypeDescription, NodeConnectionType } from 'n8n-workflow';
+
+    export class ${className} implements INodeType {
+      description: INodeTypeDescription = ${source};
+  }`;
```

Comment:
It'd be nice if `recast` supported generating TS, but bringing in `ts-morph` only for this use case would be overkill, so I can live with a bit of string manipulation.